src/components/Upload/UploadFilesForm.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { Except } from 'type-fest';
+import { DialogForm, DialogFormProps } from '../Dialog/DialogForm';
+import { DropzoneField, SubmitError } from '../form';
+import { UploadCallback, useUpload } from '../Upload';
+
+export type UploadFilesFormProps = DialogFormProps<{ files: File[] }> & {
+  callback: UploadCallback;
```

Comment:
I think this prop could be named better. Following the on* convention would be nice but I can't think of a name that fits. `finalizeUpload` seems nice.